core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -40,6 +44,8 @@ class ReplicaAlterLogDirsThread(name: String,
                                 isInterruptible = false,
                                 brokerTopicStats) {
 
+  private val assignmentRequestStates: ConcurrentHashMap[TopicPartition, DirectoryEventRequestState] = new ConcurrentHashMap()
```

Comment:
Don't we also need to consider this state when the reassignment is cancelled?

Reply:
I updated `removePartitions` which get invoked from `ReplicaManager.alterReplicaLogDirs` during cancellation to fire the revert of the assignment request if we have on queued.